### PR TITLE
DDCE-4843 - update the logic for P11D and Forecast display

### PR DESCRIPTION
**Pull request by mywyau**
ATs -> https://github.com/hmrc/tax-history-acceptance-tests/pull/181
Backend  -> https://github.com/hmrc/tax-history/pull/170
Stubs -> https://github.com/hmrc/tax-history-stubs/pull/74

### CODE REVIEW

**Comment by mywyau** [comment]
@lmugnai had to remake this branch, messed up the git history but addressed the comments. Left the new fields in the case class as is, still looking into it.

File: app/controllers/EmploymentDetailController.scala [hunk 1]
```diff
@@ -152,9 +152,9 @@ class EmploymentDetailController @Inject() (
   )(implicit hc: HeaderCarrier, request: Request[_]) = {
     val employment = dateUtils.formatEmploymentDates(empResponse.json.as[Employment])
     for {
-      payAndTax       <- getPayAndTax(nino, taxYear, employmentId)
-      companyBenefits <- getCompanyBenefits(nino, taxYear, employmentId)
-      incomeSource    <- getIncomeSource(nino, taxYear, employmentId)
+      payAndTax: Option[PayAndTax]          <- getPayAndTax(nino, taxYear, employmentId)
+      companyBenefits: List[CompanyBenefit] <- getCompanyBenefits(nino, taxYear, employmentId)
+      incomeSource: Option[IncomeSource]    <- getIncomeSource(nino, taxYear, employmentId)
```
Comment (mywyau) [inline app/controllers/EmploymentDetailController.scala:157]: add the type annotations to help make it easier to follow.

File: project/plugins.sbt [hunk 2]
```diff
@@ -10,12 +10,12 @@ ThisBuild / libraryDependencySchemes ++= Seq(
...
-addSbtPlugin("com.beautiful-scala" % "sbt-scalastyle"           % "1.5.1")
-addSbtPlugin("uk.gov.hmrc"         % "sbt-distributables"       % "2.2.0")
-addSbtPlugin("com.typesafe.play"   % "sbt-plugin"               % "2.8.20")
-addSbtPlugin("io.github.irundaia"  % "sbt-sassify"              % "1.5.2")
-addSbtPlugin("com.timushev.sbt"    % "sbt-updates"              % "0.6.3")
-addSbtPlugin("org.scalameta"       % "sbt-scalafmt"             % "2.5.2")
-addSbtPlugin("uk.gov.hmrc"         % "sbt-accessibility-linter" % "0.35.0")
+addSbtPlugin("uk.gov.hmrc"        % "sbt-auto-build"           % "3.15.0")
+addSbtPlugin("org.scoverage"      % "sbt-scoverage"            % "2.0.9")
+addSbtPlugin("org.scalastyle"     % "scalastyle-sbt-plugin"    % "1.0.0")
```
Comment (mywyau) [inline project/plugins.sbt:15]: updated the plugin to avoid the bobby warning as suggested

File: app/model/api/CompanyBenefit.scala [hunk 3]
```diff
@@ -25,9 +26,14 @@ case class CompanyBenefit(
   iabdType: String,
   amount: BigDecimal,
   source: Option[Int] = None,
+  captureDate: Option[String],
```
Comment (mywyau) [inline app/model/api/CompanyBenefit.scala:29]: investigating the field based on comments

Comment (mywyau) [reply app/model/api/CompanyBenefit.scala:29]: yup not needed think we can revert, might of been an error elsewhere when I was checking the change on the backend

File: test/connectors/TaxHistoryConnectorSpec.scala [hunk 4]
```diff
@@ -26,9 +25,11 @@ import play.api.libs.json.{JsArray, JsValue, Json}
 import support.BaseSpec
 import uk.gov.hmrc.domain.Nino
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+import uk.gov.hmrc.time.TaxYear
```
Comment (lmugnai) [inline test/connectors/TaxHistoryConnectorSpec.scala:28]: I think this can be removed now too

File: test/views/taxhistory/Constants.scala [hunk 5]
```diff
@@ -19,6 +19,7 @@ package views.taxhistory
 import model.api.EmploymentPaymentType.{JobseekersAllowance, OccupationalPension}
 import model.api._
 import models.taxhistory.Person
+import uk.gov.hmrc.time.TaxYear
```
Comment (lmugnai) [inline test/views/taxhistory/Constants.scala:22]: THink this can be removed now too

**Review by lmugnai — DISMISSED** [review]

**Comment by mywyau** [comment]
Bot warning : Reason: Play 3.0 upgrade - Deprecate [Play 2.9 and below](https://confluence.tools.tax.service.gov.uk/x/SQAoLg) 

We have upcoming work for this

**Pull request merged**